samples/glsl_flat_smooth_interpolators.html

```diff
@@ -145,43 +145,51 @@
         var uniformMvNormalLocations = [gl.getUniformLocation(programs[VIEWPORTS.LEFT], "mvNormal"), gl.getUniformLocation(programs[VIEWPORTS.RIGHT], "mvNormal")];
-        
-        var vertexPosBuffer = gl.createBuffer();
-        var vertexNormalBuffer = gl.createBuffer();
+
+        var vertexBuffer = gl.createBuffer();
         var vertexElementBuffer = gl.createBuffer();
         var vertexArray = gl.createVertexArray();
 
-        // -- Load obj then render
-        var objURL = '../assets/obj/di_model_tri.obj';
-        loadObj(objURL, function(mesh) {
+        // -- Load gltf then render
+        var gltfUrl = '../assets/gltf/di_model_tri.gltf';
+        loadGltfGeometry(gltfUrl, function(scene) {
+
             // -- Initialize buffer
-            var positions = new Float32Array(mesh.vertices); 
-            gl.bindBuffer(gl.ARRAY_BUFFER, vertexPosBuffer);
-            gl.bufferData(gl.ARRAY_BUFFER, positions, gl.STATIC_DRAW);
-            gl.bindBuffer(gl.ARRAY_BUFFER, null);
-            
-            var normals = new Float32Array(mesh.vertexNormals);
-            gl.bindBuffer(gl.ARRAY_BUFFER, vertexNormalBuffer);
-            gl.bufferData(gl.ARRAY_BUFFER, normals, gl.STATIC_DRAW);
+            var vertices = scene.vertices;
+            gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
+            gl.bufferData(gl.ARRAY_BUFFER, vertices, gl.STATIC_DRAW);
             gl.bindBuffer(gl.ARRAY_BUFFER, null);
-            
-            var indices = new Uint16Array(mesh.indices);
+
+            var indices = scene.indices;
             gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, vertexElementBuffer);
             gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, indices, gl.STATIC_DRAW);
             gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, null);
-            
+
             // -- Initialize vertex array
-            gl.bindVertexArray(vertexArray);
-            var vertexPosLocation = 0; // set with GLSL layout qualifier
-            gl.bindBuffer(gl.ARRAY_BUFFER, vertexPosBuffer);
-            gl.vertexAttribPointer(vertexPosLocation, 3, gl.FLOAT, false, 0, 0);
-            gl.enableVertexAttribArray(vertexPosLocation);
-            gl.bindBuffer(gl.ARRAY_BUFFER, null);
+            var POSITION_LOCATION = 0; // set with GLSL layout qualifier
+            var NORMAL_LOCATION = 1; // set with GLSL layout qualifier
 
-            var vertexNormalLocation = 1; // set with GLSL layout qualifier
-            gl.bindBuffer(gl.ARRAY_BUFFER, vertexNormalBuffer);
-            gl.vertexAttribPointer(vertexNormalLocation, 3, gl.FLOAT, false, 0, 0);
-            gl.enableVertexAttribArray(vertexNormalLocation);
+            gl.bindVertexArray(vertexArray);
+            gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
+            gl.vertexAttribPointer(
+                POSITION_LOCATION,
+                scene.positionNumberOfComponents,
+                gl.FLOAT,
```

Comment:
Here and below, get the type (`FLOAT` in this case) from the accessor, https://github.com/KhronosGroup/glTF/blob/master/specification/README.md#reference-accessor